packages/vitest/src/runtime/runners/test.ts

```diff
@@ -27,7 +27,17 @@ export class VitestTestRunner implements VitestRunner {
     this.snapshotClient.clear()
   }
 
-  async onAfterRunFiles() {
+  async onAfterRunFiles(files?: File[]) {
+    // mark snapshots in skipped tests as not obsolete
+    // TODO: this probably doesn't work when `VitestTestRunner` is handling multiple files concurrently,
+    //       but `snapshotClient.startCurrentRun/finishCurrentRun` doesn't work in that case already?
+    for (const test of getTests(files ?? [])) {
+      if (test.mode === 'skip') {
+        const name = getNames(test).slice(1).join(' > ')
+        this.snapshotClient.skipTestSnapshots(name)
+      }
+    }
```

Comment:
I'm assuming snapshot system currently doesn't work when there are multiple `files` running here, but I'm not able to reproduce such case yet.
My assumption might be totally wrong, but I'm investigating a little more.

Reply:
The code should be moved to `onAfterRunSuite` if `SnapshotClient.finishCurrentRun` is moved to `onAfterRunSuite` by https://github.com/vitest-dev/vitest/pull/4796.